### PR TITLE
fixed auth routing

### PR DESCRIPTION
**Pull request by Stommy123**
### Summary

- Added `/api` prefix to all backend routing

- Added "temp" fix for backend auth issues for react routes

### CODE REVIEW

File: server/middleware/authentication/index.js [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
+  ExtractJwt = require('passport-jwt').ExtractJwt,
+  jwt = require('jsonwebtoken');
 
 let jwtOptions = {
   jwtFromRequest: (req) => {
+    if (!req.url.includes('/api')) {
+      const token = jwt.sign({ react_app: true }, process.env.JWT_SECRET);
+
+      return token;
+    }
```
Comment (Stommy123) [inline server/middleware/authentication/index.js:13]: @ErnieAtLYD Hey, so this is a really dirty way i came up with to just get around the routing issue. Essentially all react routes is after `app.use(passport)`, so express is trying to validate that they're logged in. 

I'm not too familiar with serving my frontend as static files from the epxress server so i don't know what the correct solution is. Was hoping you could take another look

Comment (ErnieAtLYD) [reply server/middleware/authentication/index.js:13]: @Stommy123 Interesting. So in our "crib sheet" GitHub repo, we have this particular order in our `/server/app.js` - I have no idea if the C38 code along is in the same order:

```jsx
// Unauthenticated routes
app.use(openRoutes);

app.use(cookieParser());

//
if (process.env.NODE_ENV === 'production') {
  // Serve any static files
  app.use(express.static(path.join(__dirname, '../client/build')));
}

app.use(
  passport.authenticate('jwt', {
    session: false
  })
);

app.use(
  fileUpload({
    useTempFiles: true,
    tempFileDir: '/tmp/images'
  })
);

//  Authenticated  Routes
app.use(userRouter);
app.use(taskRouter);

if (process.env.NODE_ENV === 'production') {
  // Handle React routing, return all requests to React app
  app.get('*', (request, response) => {
    response.sendFile(path.join(__dirname, '../client/build', 'index.html'));
  });
}

module.exports = app;
```

Comment (ErnieAtLYD) [reply server/middleware/authentication/index.js:13]: @Stommy123 I'm also wondering if https://github.com/wyncode/C38_snakes_on_a_repo/blob/master/server/app.js#L16-L27 needs to be after their reference to:
```jsx
app.use(
  passport.authenticate('jwt', {
    session: false
  })
);
```
Note that we never had this code https://github.com/wyncode/C38_snakes_on_a_repo/blob/master/server/app.js#L16-L21 in our final templates or sample To Do apps, so I'm not sure where that comes from. @e-a-w, do you know where that particular line came from?

Comment (e-a-w) [reply server/middleware/authentication/index.js:13]: @ErnieAtLYD We did this part of the code first, so we can't remember exactly what we were looking off of, but it is in the C38 todo app code-along, on line 22 from [here](https://github.com/wyncode/c38_todo_app/blob/master/server/app.js), so I suspect we got it from there.

**Comment by ErnieAtLYD** [comment]
I'm good with this BUT let's hold off until after Pitchnight to merge to avoid opening up a Pandora's Box of issues

**Review by ErnieAtLYD — APPROVED** [review]

**Review by ErnieAtLYD — APPROVED** [review]

**Pull request merged**